src/tyr/actor.cc

```diff
@@ -37,13 +41,18 @@ struct actor_t::pimpl_t {
 actor_t::actor_t(const boost::property_tree::ptree& config, bool auto_cleanup)
     : pimpl(new pimpl_t(config)), auto_cleanup(auto_cleanup) {
 }
+actor_t::actor_t(const boost::property_tree::ptree& config,
+                 baldr::GraphReader& reader,
+                 bool auto_cleanup)
+    : pimpl(new pimpl_t(config, reader)), auto_cleanup(auto_cleanup) {
+}
 
 void actor_t::cleanup() {
   pimpl->cleanup();
 }
 
-valhalla::Api actor_t::unserialized_route(const std::string& request_str,
-                                          const std::function<void()>& interrupt) {
+std::string
+actor_t::route(const std::string& request_str, const std::function<void()>& interrupt, Api* api) {
```

Comment:
all the functions allow you to get back the protobuf object that was used to fulfill the request